qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/SingleBrokerDataDeletionTest.java

```diff
@@ -47,284 +47,247 @@
+    // deployment commands
+    final LogStream logStream = clusteringRule.getLogStream(1);
+    final LogStreamReader reader = logStream.newLogStreamReader().join();
     ControllableExporter.updatePosition(false);
-    ControllableExporter.RECORD_TYPE_FILTER.set(r -> r == RecordType.COMMAND);
-    ControllableExporter.VALUE_TYPE_FILTER.set(r -> r == ValueType.DEPLOYMENT);
-    writeSegments(broker, 2);
-    clusteringRule
-        .getClient()
-        .newDeployCommand()
-        .addWorkflowModel(
-            Bpmn.createExecutableProcess("process").startEvent().done(), "process.bpmn")
-        .send()
-        .join();
+    ControllableExporter.RECORD_TYPE_FILTER.set(t -> t == RecordType.COMMAND);
+    ControllableExporter.VALUE_TYPE_FILTER.set(t -> t == ValueType.DEPLOYMENT);
+
+    // when - filling up the log with messages and NO deployments
+    publishEnoughMessagesForCompaction();
+    deployDummyProcess();
     await("until at least one record is exported")
         .atMost(Duration.ofSeconds(5))
         .untilAsserted(
             () -> assertThat(ControllableExporter.EXPORTED_RECORDS).hasValueGreaterThan(0));
 
-    // enforce compaction
-    final var segmentsBeforeSnapshot = getSegmentsCount(broker);
-    clusteringRule.getClock().addTime(SNAPSHOT_PERIOD);
+    // memorize first position pre compaction to compare later on
+    reader.seekToFirstEvent();
+    final long firstPositionPreCompaction = reader.getPosition();
 
-    // then
-    assertThat(clusteringRule.waitForSnapshotAtBroker(broker)).isNotNull();
-    await()
+    // then - enforce compaction and make sure we have less records than we previously did
+    clusteringRule.getClock().addTime(SNAPSHOT_PERIOD);
+    clusteringRule.waitForSnapshotAtBroker(clusteringRule.getBroker(0));
+    await("until some data before the deployment command was compacted")
         .untilAsserted(
-            () ->
-                assertThat(getSegmentsCount(broker))
-                    .describedAs("Expected less segments after a snapshot is taken")
-                    .isLessThan(segmentsBeforeSnapshot));
+            () -> {
+              reader.seekToFirstEvent();
+              final long firstPositionPostCompaction = reader.getPosition();
+              assertThat(firstPositionPostCompaction).isGreaterThan(firstPositionPreCompaction);
+              assertContainsDeploymentCommand(reader);
+            });
   }
 
   @Test
   public void shouldNotCompactUnacknowledgedEventsEvenIfSkipping() {
-    // given
-    final RecordMetadata metadata = new RecordMetadata();
-    final Broker broker = clusteringRule.getBroker(0);
-    final var logstream = clusteringRule.getLogStream(1);
-    final var reader = logstream.newLogStreamReader().join();
-
-    // when
+    // given - an exporter which does not update its own position and only accepts deployment
+    // commands
+    final LogStream logStream = clusteringRule.getLogStream(1);
```

Comment:
Replace the magic number with a constant. Or, extract `clusteringRule.getLogStream(1).newLogStreamReader().join()` into a new method.

```suggestion
    final LogStream logStream = clusteringRule.getLogStream(PARTITION_ID);
```